gpflow/multioutput/features.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2018 GPflow authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from .kernels import Mok, SharedIndependentMok, SeparateIndependentMok, SeparateMixedMok
+from ..features import InducingPoints, InducingFeature, Kuu, Kuf
+from ..dispatch import dispatch
+from .. import settings
+
+float_type = settings.float_type
+
+class Mof(InducingFeature):
+    """
+    Class used to indicate that we are dealing with
+    features that are used for multiple outputs.
+    """
+    pass
+
+
+class SharedIndependentMof(Mof):
+    """
+    Same feature is used for each output.
+    """
+    def __init__(self, feat):
+        Mof.__init__(self)
+        self.feat = feat
+
+    def __len__(self):
+        return len(self.feat)
+
+
+class SeparateIndependentMof(Mof):
+    """
+    A different feature is used for each output.
+    Note: each feature should have the same number of points, M.
+    """
+    def __init__(self, feat_list):
+        Mof.__init__(self)
+        self.feat_list = feat_list
+
+    def __len__(self):
+        return len(self.feat_list[0])
+
+
+class MixedKernelSharedMof(SharedIndependentMof):
+    """
+    This Mof is used in combination with the `SeparateMixedMok`.
+    Using this feature with the `SeparateMixedMok` leads to the most efficient code.
+    """
+    pass
+
+
+# ---------------------------------------------------------------------
+################################ Kuf ##################################
+# ---------------------------------------------------------------------
+
+@dispatch(InducingPoints, Mok, object)
+def Kuf(feat, kern, Xnew):
+    return kern.K(feat.Z, Xnew, full_cov_output=True)  #  M x P x N x P
+
+
+@dispatch(SharedIndependentMof, SharedIndependentMok, object)
+def Kuf(feat, kern, Xnew):
+    return Kuf(feat.feat, kern.kern, Xnew)  # M x N
+
+
+@dispatch(SeparateIndependentMof, SharedIndependentMok, object)
+def Kuf(feat, kern, Xnew):
+    return tf.stack([Kuf(f, kern.kern, Xnew) for f in feat.feat_list], axis=0)  # L x M x N
+
+
+@dispatch(SharedIndependentMof, SeparateIndependentMok, object)
+def Kuf(feat, kern, Xnew):
+    return tf.stack([Kuf(feat.feat, k, Xnew) for k in kern.kern_list], axis=0)  # L x M x N
```

Comment:
It seems that in master branch Combination kernel is no longer using kern_list and this makes error.